kafka-connect-gcp-storage/src/main/scala/io/lenses/streamreactor/connect/gcp/storage/sink/GCPStorageWriterManagerCreator.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.storage.sink
+
+import cats.implicits.catsSyntaxEitherId
+import com.typesafe.scalalogging.LazyLogging
+import io.lenses.streamreactor.connect.cloud.common.config.ConnectorTaskId
+import io.lenses.streamreactor.connect.cloud.common.formats
+import io.lenses.streamreactor.connect.cloud.common.formats.writer.FormatWriter
+import io.lenses.streamreactor.connect.cloud.common.model.Offset
+import io.lenses.streamreactor.connect.cloud.common.model.Topic
+import io.lenses.streamreactor.connect.cloud.common.model.TopicPartition
+import io.lenses.streamreactor.connect.cloud.common.model.location.CloudLocation
+import io.lenses.streamreactor.connect.cloud.common.sink.FatalCloudSinkError
+import io.lenses.streamreactor.connect.cloud.common.sink.SinkError
+import io.lenses.streamreactor.connect.cloud.common.sink.commit.CommitPolicy
+import io.lenses.streamreactor.connect.cloud.common.sink.config.PartitionField
+import io.lenses.streamreactor.connect.cloud.common.sink.naming.KeyNamer
+import io.lenses.streamreactor.connect.cloud.common.sink.seek.IndexManager
+import io.lenses.streamreactor.connect.cloud.common.sink.transformers.TopicsTransformers
+import io.lenses.streamreactor.connect.cloud.common.sink.writer.WriterManager
+import io.lenses.streamreactor.connect.cloud.common.storage.StorageInterface
+import io.lenses.streamreactor.connect.gcp.storage.sink.config.GCPStorageSinkConfig
+import io.lenses.streamreactor.connect.gcp.storage.sink.config.SinkBucketOptions
+import io.lenses.streamreactor.connect.gcp.storage.storage.GCPStorageFileMetadata
+
+import java.io.File
+import scala.collection.immutable
+
+object GCPStorageWriterManagerCreator extends LazyLogging {
```

Comment:
Is there much difference to the other writer creators? seems most code relies on bucket-for-topic mapping and that's the same across all cloud storage configs. Maybe we can avoid duplicating code even more

Reply:
💯  - these 3 have become 1 and it was much less painful than expected.  Thanks!